### PR TITLE
Code Coverage information

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Add CodeCov publishing for our windows builds.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -747,6 +767,15 @@ Task("Publish-NuGet")
     Information("Publish-NuGet Task failed, but continuing with next Task...");
 });
 
+Task("Publish-Coverage")
+    .IsDependentOn("Run-Unit-Tests")
+    .WithCriteria(() => isRunningOnAppVeyor)
+    .Does(() =>
+{
+    CoverallsNet(artifactsDir.GetFilePath(new FilePath("./coverage.xml")), CoverallsNetReportType.OpenCover,
+        new CoverallsNetSettings {RepoToken = "jKaVvndtCVJ2y7pNwQtYeHPn2jAxUpuyh", UseRelativePaths = true, ServiceName = "AppVeyor"});
```
Comment (wieslawsoltes) [inline build.cake]: This should be set as env variable like: COVERALLS_REPO_TOKEN and probably as secure?

File: build.cake [hunk 2]
```diff
@@ -747,6 +767,15 @@ Task("Publish-NuGet")
     Information("Publish-NuGet Task failed, but continuing with next Task...");
 });
 
+Task("Publish-Coverage")
+    .IsDependentOn("Run-Unit-Tests")
+    .WithCriteria(() => isRunningOnAppVeyor)
```
Comment (wieslawsoltes) [inline build.cake]: I would add some additional criteria so we do not push coverage for PR and only from main repo and branch:

``` csharp
    .WithCriteria(() => !isLocalBuild)
    .WithCriteria(() => !isPullRequest)
    .WithCriteria(() => isMainRepo)
    .WithCriteria(() => isMasterBranch)
```

File: build.cake (hunk 1 above)
Comment (jkoritzinsky) [inline build.cake]: Probably. I'll make that change later today.

File: build.cake (hunk 2 above)
Comment (jkoritzinsky) [inline build.cake]: We want PR coverage, at least if we do CodeCov. CodeCov provides a GitHub PR check for how coverage changes. I don't know what Coveralls provides in that department.

**Comment by stevepeak** [comment]
Codecov would be honored to have your business. However, it appears the upload size of this report far exceeds our largest report seen so far. The previous record was 56mb, yours is 135mb. Codecov, at this time, will not be able to process what size file. Is there any way of reducing the file size by stripping out unnecessary contents?

**Comment by jkoritzinsky** [comment]
@stevepeak I'll take another look at stripping out some more unneeded coverage information.

**Comment by jkoritzinsky** [comment]
Code coverage file is now much smaller and uploading to Codecov.io!

**Comment by jkoritzinsky** [comment]
@grokys Can you take a look? This is ready to be merged in.

**Comment by grokys** [comment]
LGTM! This will be really useful info, thanks for sorting this out Jeremy!

**Pull request merged**